mutiny-core/src/nostr/nwc.rs

```diff
@@ -1032,26 +1135,319 @@ impl NostrWalletConnect {
+
+        for param in params.invoices {
+            let pay_invoice_request = PayInvoiceRequest {
+                params: param,
+                is_multi_pay: true,
+            };
+            match self
+                .handle_nwc_invoice_payment(
+                    event.clone(),
+                    node,
+                    nostr_manager,
+                    pay_invoice_request,
+                    needs_delete,
+                    needs_save,
+                )
+                .await
+            {
+                Ok(Some(response_event)) => {
+                    if let Err(e) = nostr_manager
+                        .client
+                        .send_event(response_event.clone())
+                        .await
+                    {
+                        log_error!(
+                            nostr_manager.logger,
+                            "Error sending multi_pay_invoice nwc response event: {}",
+                            e
+                        )
+                    } else {
+                        response_events.push(response_event);
+                    }
+                }
+                Ok(None) => continue,
+                Err(e) => {
+                    log_error!(
+                        nostr_manager.logger,
+                        "Error handling multi_pay_invoice request: {}",
+                        e
+                    )
+                }
+            }
+        }
+
+        if response_events.is_empty() {
+            Ok(None)
+        } else {
+            Ok(Some(NwcResponse::MultiEvent(response_events)))
+        }
+    }
+
+    async fn handle_nwc_keysend_payment<S: MutinyStorage, P: PrimalApi, C: NostrClient>(
+        &mut self,
+        event: Event,
+        node: &impl InvoiceHandler,
+        nostr_manager: &NostrManager<S, P, C>,
+        params: PayKeysendRequestParams,
+        is_multi_keysend: bool,
+    ) -> anyhow::Result<Option<Event>> {
+        let method = if is_multi_keysend {
+            Method::MultiPayKeysend
+        } else {
+            Method::PayKeysend
+        };
 
-                let p_tag = Tag::public_key(event.pubkey);
-                let e_tag = Tag::event(event.id);
+        match self.profile.spending_conditions.clone() {
+            SpendingConditions::SingleUse(_) => {
+                // ignore this case since keysend payment requests are not generated from mutiny gifts
+                Ok(None)
+            }
+            SpendingConditions::RequireApproval => {
+                // respond with payment failed error
+                // only do keysend payments for budgeted profiles
+                let content = Response {
+                    result_type: method,
+                    error: Some(NIP47Error {
+                        code: ErrorCode::PaymentFailed,
+                        message: String::from("not enough balance to make payment"),
```

Comment:
I think it'd be better to do an `UNAUTHORIZED` here with a better error string about the budget

Reply:
`"Keysend only supported for profiles with a budget"`